File: solutions/Javascript/Range Sum of BST.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var rangeSumBST = function (root, low, high) {
  let res = [0];
  range(root, low, high, res);
  return res[0];
};

function range(root, low, high, sum) {
  if (root) {
    if (root.val >= low && root.val <= high) {
      sum[0] = sum[0] + root.val;
    }
    range(root.left, low, high, sum);
    range(root.right, low, high, sum);
  }
}

File: solutions/python/Range Sum of BST.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        #DFS
        sum=0
        queue=[]
        queue.append(root)
        while len(queue)>0:
            node=queue.pop(0)
            if node.val>=low and node.val<=high:
                sum=sum+node.val
            if node.left:
                queue.append(node.left)
            if node.right:
                queue.append(node.right)
        return sum
        #----------------------
        #BFS
        # sum=0
        # stack=[]
        # while(root or len(stack)>0):
        #     while(root):
        #         stack.append(root)
        #         root=root.left
        #     node=stack.pop()
        #     if node.val>=low and node.val<=high:
        #         sum=sum+node.val
        #     root=node.right
        # return sum


